docker.js

```diff
@@ -0,0 +1,174 @@
+const _ = require("lodash");
+
+function createVolumeConfig(path) {
+  if (!path) {
+    throw new Error("Path is required to create Volume Config.");
+  }
+  if (!_.isString(path)) {
+    throw new Error("Path parameter must be a string.");
+  }
+
+  const pathEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPointEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPoint = generateRandomTemporaryPath();
+
+  return {
+    path: {
+      environmentVariable: {
+        name: pathEnvironmentVariableName,
+        value: path,
+      },
+      value: `$${pathEnvironmentVariableName}`,
+    },
+    mountPoint: {
+      environmentVariable: {
+        name: mountPointEnvironmentVariableName,
+        value: mountPoint,
+      },
+      value: `$${mountPointEnvironmentVariableName}`,
+    },
+  };
+}
+
+function sanitizeCommand(command) {
+  if (!command || !_.isString(command)) {
+    throw new Error("Command parameter must be a string.");
+  }
+
+  return `sh -c ${JSON.stringify(command)}`;
+}
+
+function mergeVolumeConfigsEnvironmentVariables(volumeConfigs) {
+  if (!volumeConfigs || !_.isArray(volumeConfigs)) {
+    throw new Error("Volume Configs parameter must be an array of objects.");
+  }
+
+  const requiredByDocker = volumeConfigs.reduce((acc, curr) => {
+    assertVolumeConfigPropertiesExistence(curr, [
+      "mountPoint.environmentVariable.name",
+      "mountPoint.environmentVariable.value",
+    ]);
+
+    return {
+      ...acc,
+      [curr.mountPoint.environmentVariable.name]: curr.mountPoint.environmentVariable.value,
+    };
+  }, {});
+
+  const requiredByShell = volumeConfigs.reduce((acc, curr) => {
+    assertVolumeConfigPropertiesExistence(curr, [
+      "path.environmentVariable.name",
+      "path.environmentVariable.value",
+    ]);
+
+    return {
+      ...acc,
+      [curr.path.environmentVariable.name]: curr.path.environmentVariable.value,
+    };
+  }, requiredByDocker);
+
+  return {
+    namesRequiredByDocker: Object.keys(requiredByDocker),
```

Comment:
This introduces inconsistency. If this function returns env vars, then it should return env vars, not env vars and their names.